### PR TITLE
Improve styling of nav, dark mode, and inspector panel

### PR DESCRIPTION
**Pull request by charliemcgrady**
* Add Infima for consistent styling with Overture docs
* Introduce dark mode for non-map components
* Match the table layout in inspector panel with docs
* Match the nav style and layout with docs
* Organize nav components into a single directory

Screenshot 2024-05-19 at 8 43 48 PM
Screenshot 2024-05-19 at 8 44 10 PM

### CODE REVIEW

**Comment by charliemcgrady** [comment]
A couple of follow ups to discuss with the group:
1. I selectively brought in Infima and a couple of styles to match the Overture docs. Specifically, the nav should be mostly consistent. Should we use Docusaurus? A shared React component library? Is Infima intended/ready to be used elsewhere?
2. Some of the diff is from me applying default prettier styles. I plan on adding it more formally and reformatting everything in a single follow up commit to avoid needless diffs in unrelated CRs. Wanted to make sure folks are aligned.

File: site/src/DarkModeToggle.jsx
Comment (charliemcgrady) [inline site/src/DarkModeToggle.jsx:1]: Moved to dedicated nav directory

**Comment by charliemcgrady** [comment]
Looking for UX feedback on the spacing with the table in the inspector panel. The screenshots above are the default Infima table spacing. I'm leaning towards compressing the table a bit to minimize scrolling / making sure as many properties are visible as possible.

**Comment by charliemcgrady** [comment]
Should we hide the download button on mobile?

**Comment by Bonkles** [comment]
Am very much aligned with getting prettier going for file formats. Infinima / related work to make things look like the other sites are also a great idea.

**Review by Bonkles — APPROVED** [review]
These changes look good to me so far! 

Q: I don't see any mentions of infinima past its inclusion in the package.json. Is that intensional? 

It'd be neat to fail the build if we have eslint errors, so that github deploys would fail early.

**Comment by charliemcgrady** [comment]
> These changes look good to me so far!
> 
> Q: I don't see any mentions of infinima past its inclusion in the package.json. Is that intensional?
> 
> It'd be neat to fail the build if we have eslint errors, so that github deploys would fail early.

Infima is a CSS framework, so the only reference are in the class names. For example, the download button uses "button button--primary" classes which will style it to be a primary button.

**Pull request merged**